models/svc/vits/vits.py

```diff
@@ -0,0 +1,295 @@
+        self.p_dropout = cfg.model.vits.p_dropout
+        self.ssl_dim = cfg.model.vits.ssl_dim
+        self.n_flow_layer = cfg.model.vits.n_flow_layer
+        self.gin_channels = cfg.model.vits.gin_channels
+        self.n_speakers = cfg.model.vits.n_speakers
+        
+        # f0
+        self.n_bins = cfg.preprocess.pitch_bin
+        self.f0_min = cfg.preprocess.f0_min
+        self.f0_max = cfg.preprocess.f0_max
+        
+        # TODO: sort out the config
+        self.cfg.model.condition_encoder.f0_min = self.cfg.preprocess.f0_min
+        self.cfg.model.condition_encoder.f0_max = self.cfg.preprocess.f0_max
+        self.condition_encoder = ConditionEncoder(self.cfg.model.condition_encoder)
+        
+        self.emb_g = nn.Embedding(self.n_speakers, self.gin_channels)
+        self.emb_uv = nn.Embedding(2, self.hidden_channels)
+
+        self.pre = nn.Conv1d(self.ssl_dim, self.hidden_channels, kernel_size=5, padding=2)
+
+        self.enc_p = ContentEncoder(
+            self.inter_channels,
+            self.hidden_channels,
+            filter_channels=self.filter_channels,
+            n_heads=self.n_heads,
+            n_layers=self.n_layers,
+            kernel_size=self.kernel_size,
+            p_dropout=self.p_dropout
+        )
+
+        assert cfg.model.generator in ['bigvgan', 'hifigan', 'melgan', 'nsfhifigan', 'apnet']
+        self.dec_name = cfg.model.generator
+        temp_cfg = copy.deepcopy(cfg)
+        temp_cfg.preprocess.n_mel = self.inter_channels
+        if cfg.model.generator == 'bigvgan':
+            temp_cfg.model.bigvgan = cfg.model.generator_config.bigvgan
+            self.dec = BigVGAN(temp_cfg)
+        elif cfg.model.generator == 'hifigan':
+            temp_cfg.model.hifigan = cfg.model.generator_config.hifigan
+            self.dec = HiFiGAN(temp_cfg)
+        elif cfg.model.generator == 'melgan':
+            temp_cfg.model.melgan = cfg.model.generator_config.melgan
+            self.dec = MelGAN(temp_cfg)
+        elif cfg.model.generator == 'nsfhifigan':
+            temp_cfg.model.nsfhifigan = cfg.model.generator_config.nsfhifigan
+            self.dec = NSFHiFiGAN(temp_cfg) # TODO: nsf need f0
+        elif cfg.model.generator == 'apnet':
+            temp_cfg.model.apnet = cfg.model.generator_config.apnet
+            self.dec = APNet(temp_cfg)
+        
+        self.enc_q = PosteriorEncoder(
+            self.spec_channels, 
+            self.inter_channels, 
+            self.hidden_channels, 
+            5, 
+            1, 
+            16, 
+            gin_channels=self.gin_channels
+        )
+        
+        self.flow = ResidualCouplingBlock(
+            self.inter_channels, 
+            self.hidden_channels, 
+            5, 
+            1, 
+            self.n_flow_layer, 
+            gin_channels=self.gin_channels,
+        )
+        
+    def forward(self, data):
+        # c, f0, uv, spec, g=None, c_lengths=None, spec_lengths=None, vol = None
```

Comment:
Use [autoDocstring](https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring) to format the function comments.